test/utils/test_utils.py

```diff
@@ -59,3 +60,53 @@ def test_validate_controls():
         validate_control_variables([arr], [arr[:1]])
     u_mod = validate_control_variables([arr], [arr], trim_last_point=True)
     assert u_mod[0].n_time == 1
+
+
+@pytest.mark.parametrize(
+    ["thresholder", "expected"], [("l0", 4), ("l1", 6), ("l2", 10)]
+)
+def test_get_regularization_1d(thresholder, expected):
+    data = np.array([[0, 1, 2]]).T
+    lam = 2
+
+    reg = get_regularization(thresholder)
+    result = reg(data, lam)
+    assert result == expected
+
+
+@pytest.mark.parametrize(
+    ["thresholder", "expected"], [("l0", 8), ("l1", 10), ("l2", 14)]
+)
+def test_get_regularization_2d(thresholder, expected):
+    data = np.array([[0, 1, 2], [1, 1, 0]]).T
+    lam = 2
+
+    reg = get_regularization(thresholder)
+    result = reg(data, lam)
+    assert result == expected
+
```

Comment:
is `reg=get_regularization()` ever called for matrix norms?  If so, just check that `reg(data, lam)` equals `reg(data.flatten(), lam)`.  That makes it clearer to a reader why the expected value is, in fact, expected.

Reply:
If n_target is more than 1, regularization will be called with a matrix.